common/scala/src/main/scala/whisk/core/entity/Identity.scala

```diff
@@ -74,6 +74,53 @@ object Identity extends MultipleReadersSingleWriterCache[Identity, DocInfo] with
         })
     }
 
+    /**
+      * Many subjects can be belonged to only one namespace
+      * Retrieves a key for namespace + subject.
+      *
+      */
+    def get(datastore: AuthStore, namespace: EntityName, subject: Subject)(
```

Comment:
we don't need a view to look up by subject - it is a primary index.

i suspect though i did not try, that you can do get on AuthStore to fetch the subject, and parameterize that on the type of a subject record. i'll create a gist for you.